Panic output vault provider terraform

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
```
Terraform v1.5.4
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.75.1
+ provider registry.terraform.io/hashicorp/google-beta v4.75.1
+ provider registry.terraform.io/hashicorp/helm v2.10.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.22.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/vault v3.18.0
```


### Panic Output
```
Stack trace from the terraform-provider-vault_v3.18.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1bc6259]

goroutine 81 [running]:
github.com/hashicorp/terraform-provider-vault/internal/provider.NewProviderMeta(0xc000b1bf80)
        github.com/hashicorp/terraform-provider-vault/internal/provider/meta.go:279 +0xbb9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc000280240, {0x244df70, 0xc000524540}, 0xd?)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/provider.go:287 +0x18f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0007f0870, {0x244dec8?, 0xc000220900?}, 0xc0009cc4c8)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:557 +0x345
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc0009945a0, {0x244df70?, 0xc00124cb40?}, 0xc000220600)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:555 +0x2ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x1fe3ee0?, 0xc0009945a0}, {0x244df70, 0xc00124cb40}, 0xc0003a38f0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000261e0, {0x2452f20, 0xc000b82ea0}, 0xc000d38ea0, 0xc0005d5f20, 0x2d47958, 0x0)
        google.golang.org/grpc@v1.50.0/server.go:1318 +0xb2b
google.golang.org/grpc.(*Server).handleStream(0xc0000261e0, {0x2452f20, 0xc000b82ea0}, 0xc000d38ea0, 0x0)
        google.golang.org/grpc@v1.50.0/server.go:1659 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.0/server.go:955 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.0/server.go:953 +0x28a

Error: The terraform-provider-vault_v3.18.0_x5 plugin crashed!
```

### Expected Behavior
Only thing I'm doing with vault provider is reading a single kv secret

### Actual Behavior
Crash

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important facts
Cluster where vault is running has been acting strange today in regards to network, and I'm working on mobile hotspot, so this may be related to connectivity to vault.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## Unreleased
 
+BUGS:
+* Fixes a panic that can occur when Vault [lookup-self](https://developer.hashicorp.com/vault/api-docs/auth/token#lookup-a-token-self) API returns nil token info ([#1978](https://github.com/hashicorp/terraform-provider-vault/pull/1978))
+
 ## 3.19.0 (Aug 2, 2023)
 FEATURES:
 * Add support for User ID configuration for PKI Secrets Engine: ([#1936](https://github.com/hashicorp/terraform-provider-vault/pull/1936))

--- a/internal/provider/meta.go
+++ b/internal/provider/meta.go
@@ -273,6 +273,9 @@ func NewProviderMeta(d *schema.ResourceData) (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to lookup token, err=%w", err)
 	}
+	if tokenInfo == nil {
+		return nil, fmt.Errorf("no token information returned from self lookup")
+	}
 
 	warnMinTokenTTL(tokenInfo)
 

